components/mediators/datamapper/org.wso2.carbon.mediator.datamapper/src/main/java/org/wso2/carbon/mediator/datamapper/DataMapperMediator.java

```diff
@@ -757,4 +749,27 @@ private ScriptEngine getScriptExecutor() {
         }
         return new ScriptEngineManager().getEngineByName(DataMapperEngineConstants.GRAALJS_ENGINE_NAME);
     }
+
+    private void initializeInputOutputType() {
+        if (inputType == null) {
+            Schema inputSchema = mappingResource.getInputSchema();
+            if (inputSchema != null && inputSchema.getSchemaMap() != null &&
+                    inputSchema.getSchemaMap().containsKey(INPUT_TYPE)) {
+                inputType = inputSchema.getSchemaMap().get(INPUT_TYPE).toString();
+            } else {
+                log.error("Input type is not defined in the input schema/synapse configuration");
```

Comment:
Earlier we consider xml as default if not defined. Now we throw error. Can this break some old configs ?

Reply:
In the DM factory[1], providing input and output types is mandatory. Hence, it is impossible to create a data mapper mediator without providing those values in the old configs. So old configs would continue to function as it is. 

[1] - https://github.com/wso2/carbon-mediation/blob/v4.7.210/components/mediators/datamapper/org.wso2.carbon.mediator.datamapper/src/main/java/org/wso2/carbon/mediator/datamapper/config/xml/DataMapperMediatorFactory.java#L109-L118